templates/consul.sysconfig.erb

```diff
@@ -0,0 +1 @@
+RPC_ADDR=<%= "-rpc-addr=%{::ipaddress_lo}:#{@rpc_port}" -%>
```

Comment:
`::ipaddress_lo` is a concern because you can configure `rpc` address in consul as well.

So in `init.pp` you need another conditional:
```
if $config_hash_real['rpc'] {
  $rpc_addr = $config_hash_real['rpc']
} elsif $config_hash_real['client_addr'] {
  $rpc_addr = $config_hash_real['client_addr']
} else {
  $rpc_addr = '127.0.0.1'
}
```

Then remove this file entirely `consul.sysconfig.erb` and use these params directly in `consul.sysv.erb like so:
`"$CONSUL" leave <%= @rpc_addr %>:<%= @rpc_port %>`

This way we can use this with other init scripts as well.

https://www.consul.io/docs/agent/options.html#rpc